workflow-core/src/commonMain/kotlin/com/squareup/workflow1/WorkflowIdentifier.kt

```diff
@@ -146,15 +146,14 @@ public class WorkflowIdentifier internal constructor(
 /**
  * The [WorkflowIdentifier] that identifies this [Workflow].
  */
-public val Workflow<*, *, *>.identifier: WorkflowIdentifier
-  get() {
-    val maybeImpostor = this as? ImpostorWorkflow
-    return WorkflowIdentifier(
-      type = Snapshottable(this::class),
-      proxiedIdentifier = maybeImpostor?.realIdentifier,
-      description = maybeImpostor?.let { it::describeRealIdentifier }
-    )
-  }
+public fun Workflow<*, *, *>.computeIdentifier(): WorkflowIdentifier {
+  val maybeImpostor = this as? ImpostorWorkflow
+  return WorkflowIdentifier(
+    type = Snapshottable(this::class),
+    proxiedIdentifier = maybeImpostor?.realIdentifier,
+    description = maybeImpostor?.let { it::describeRealIdentifier }
```

Comment:
The reason we use a lazy delegate to initialize `identifier` in `StatelessWorkflow` and `StatefulWorkflow` abstract classes is because `computeIdentifier` does reference the `realIdentifier` of the impostor, which won't be initialized yet at the time that the abstract class initializer happens.

I guess I should probably document that..